npm/ng-packs/packages/oauth/src/lib/strategies/auth-flow-strategy.ts

```diff
@@ -80,6 +84,28 @@ export abstract class AuthFlowStrategy {
       .catch(this.catchError);
   }
 
+  protected navigateToPreviousUrl() {
+    this.oAuthService.events
```

Comment:
it should be disposed.

Reply:
I'm not sure how can I dispose this method. Other examples seems like same

https://github.com/abpframework/abp/blob/419c2f2af451691ad15b6b4d834c3663b141288c/npm/ng-packs/packages/oauth/src/lib/strategies/auth-flow-strategy.ts#L69

# 

I just improved with condition

![image](https://github.com/abpframework/abp/assets/49063256/3b74ff0c-5729-4689-aac2-80d1e2d53c52)

Reply:
https://rxjs.dev/api/operators/take
.take(1) before the subscribe.  it turn to promise-like obserable.